keps/sig-api-machinery/5366-graceful-leader-transition/README.md

```diff
@@ -0,0 +1,940 @@
+cleared/stopped when the leader lock is lost.
+
+Scenarios:
+
+- Component acquires leader lock should start its control loop
+- Component releasing leader lock should shut down its control loop
+- Component acquiring and releasing the leader lock multiple times should not leak memory
+- Component acquiring and releasing the leader lock multiple times should have ONLY ONE control loop running
+- Components in follower mode should not start control loop or otherwise allocate unnecessary memory
+- Only one control loop should be running at all times
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+This is primarily testing for leader election's interaction with components, and will be tested via integration and e2e tests.
+
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+See the above scenarios for test plan.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+See the above scenarios for test plan.
+
+### Graduation Criteria
+
+#### Alpha
```

Comment:
alpha needs to include testing for zero leaking go routines.

Reply:
Added